### ISSUE TITLE
0.4.1: AttributeError: 'tuple' object has no attribute 'tagClass'

### ISSUE THREAD

**Issue by reox** [etingof/pyasn1#106]
Again some issues with my certificate parsing. You can use the same code from https://github.com/etingof/pyasn1/issues/67

```
Traceback (most recent call last):
  File "certtest.py", line 10, in <module>
    cert = encode(message[1][3])
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/cer/encoder.py", line 215, in encodeValue
    substrate += encodeFun(value[idx], asn1Spec, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/cer/encoder.py", line 179, in encodeValue
    for x in value]
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/cer/encoder.py", line 179, in <listcomp>
    for x in value]
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/codec/ber/encoder.py", line 182, in encodeValue
    asn1Spec = value.clone(tagSet=tag.TagSet(tagSet.baseTag, tagSet.baseTag))
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/type/tag.py", line 189, in __init__
    [(superTag.tagClass, superTag.tagId) for superTag in superTags]
  File "/home/vagrant/.local/lib/python3.5/site-packages/pyasn1/type/tag.py", line 189, in <listcomp>
    [(superTag.tagClass, superTag.tagId) for superTag in superTags]
AttributeError: 'tuple' object has no attribute 'tagClass'
```

This error happens since 10 minutes ago as 0.4.1 was published to pypi. In 0.3.7 everything was working,

**Comment by etingof** [comment]
Thank you for the heads up!

Could you please try [this fix](https://github.com/etingof/pyasn1/pull/107)?

**Comment by reox** [comment]
Works for me!

**Comment by KingjZhang** [comment]
```"/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/cer/encoder.py", line 215, in encodeValue
    substrate += encodeFun(value[idx], asn1Spec, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/cer/encoder.py", line 179, in encodeValue
    for x in value]
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/cer/encoder.py", line 179, in <listcomp>
    for x in value]
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 423, in encodeValue
    chunk = encodeFun(component, asn1Spec, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 636, in __call__
    substrate = concreteEncoder.encode(value, asn1Spec, self, **options)
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 85, in encode
    value, asn1Spec, encodeFun, **options
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/codec/ber/encoder.py", line 182, in encodeValue
    asn1Spec = value.clone(tagSet=tag.TagSet(tagSet.baseTag, tagSet.baseTag))
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/type/tag.py", line 189, in __init__
    [(superTag.tagClass, superTag.tagId) for superTag in superTags]
  File "/home/king/.pyenv/versions/general/lib/python3.6/site-packages/pyasn1/type/tag.py", line 189, in <listcomp>
    [(superTag.tagClass, superTag.tagId) for superTag in superTags]
AttributeError: 'tuple' object has no attribute 'tagClass'
```
I got same issue on Ubuntu 16.04 and Python 3.6.3 for pyasn1 V0.4.1

**Comment by etingof** [comment]
I've refactored the fix published it as 0.4.2. I am not quite satisfied with it so I will probably rework it later. 

But it should hopefully fix your immediate problem. Does it?

**Comment by reox** [comment]
I upgraded to 0.4.2 and it works fine! thanks a lot!
